src/common.jl

```diff
@@ -37,5 +54,111 @@ function plot_map(gdf::DataFrame; geom_col::Symbol=:geometry)
+        for i in 1:size(x,1)
+        reef_poly = x[i,:]
+        intersecting = DataFrame(GBRMPA_ID = [], area_ID = [], inter_area = [])
+
+            for j in 1:size(y,1)
+            interest_area = y[j,:]
+
+                if AG.intersects(reef_poly.geometry, interest_area[ygeom_col])
+                    inter_area = AG.intersection(reef_poly.geometry, interest_area[ygeom_col])
+                    inter_area = AG.geomarea(inter_area)
+                    prop_area = inter_area/AG.geomarea(reef_poly.geometry)
+                    if prop_area >= 0.5
+                        push!(intersecting, [x[i, xid], y[j, yid],inter_area])
+                    else push!(intersecting, [ missing , missing, missing])
+                    end
+                else push!(intersecting, [ x[i,xid] , missing, missing])
+                end
+            end
+            if all(ismissing,intersecting.area_ID)
+                push!(rel_areas, [intersecting[1,xid], intersecting[1,:area_ID]])
+            else
+                dropmissing!(intersecting)
+                push!(rel_areas, [intersecting[findmax(intersecting.inter_area)[2],xid],
+                intersecting[findmax(intersecting.inter_area)[2],:area_ID]])
+            end
+        end
+    else
+        for i in 1:size(x,1)
+            reef_poly = x[i,:]
+            intersecting = DataFrame(GBRMPA_ID = [], area_ID = [], inter_area = [])
+
+            for j in 1:size(y,1)
+                interest_area = y[j,:]
+
+                if AG.intersects(reef_poly.geometry, interest_area[ygeom_col])
+                    inter_area = AG.intersection(reef_poly.geometry, interest_area[ygeom_col])
+                    inter_area = AG.geomarea(inter_area)
+                    prop_area = inter_area/AG.geomarea(reef_poly.geometry)
+                    push!(intersecting, [x[i, xid], y[j, yid],prop_area])
+
+                else push!(intersecting, [ x[i,xid] , missing, missing])
+                end
+            end
+            if all(ismissing,intersecting.area_ID)
+                push!(rel_areas, [intersecting[1,xid], intersecting[1,:area_ID]])
+            else
+                dropmissing!(intersecting)
+                push!(rel_areas, [intersecting[findmax(intersecting.inter_area)[2],xid],
+                intersecting[findmax(intersecting.inter_area)[2],:area_ID]])
+            end
+        end
+    end
+    return rel_areas
+end
+
+"""
+    find_latest_file(dir::String)
+    Function that finds the latest file in a directory with the date format YYYY-mm-dd-THH-SS.
+    Intended to find the latest rrap_shared_lookup output file for input into the next script.
+
+# Arguments
+- `dir` : Target directory string.
+"""
+function find_latest_file(dir::String)
+    files = readdir(dir)
+    files = files[contains.(files, "rrap_shared_lookup")]
+    timestamps = map(f -> Dates.DateTime(chop(f, head=19, tail=5), "YYYY-mm-dd-THH-SS"), files)
+    latest = files[argmax(timestamps)]
+    return latest
```

Comment:
There's a few exceptions to this rule but I prefer return statements to be separate for clarity.

```suggestion
    latest = files[argmax(timestamps)]

    return latest
```